src/include/common/vector/auxiliary_buffer.h

```diff
@@ -43,6 +44,23 @@ class StructAuxiliaryBuffer : public AuxiliaryBuffer {
     std::vector<std::shared_ptr<ValueVector>> childrenVectors;
 };
 
+struct ArrowData {
+    common::offset_t startOffset;
+    std::shared_ptr<arrow::RecordBatch> recordBatch;
+
+    ArrowData() = default;
+
+    ArrowData(common::offset_t startOffset, std::shared_ptr<arrow::RecordBatch> recordBatch)
+        : startOffset{startOffset}, recordBatch{std::move(recordBatch)} {}
+};
+
+class ArrowDataAuxiliaryBuffer : public AuxiliaryBuffer {
+    friend class ArrowDataVector;
+
+private:
+    ArrowData arrowData;
```

Comment:
I would not create ArrowData class and just save offset and recordBatch here. If you do want to use ArrowData, use with unique_ptr. 

Also why is recordBatch a shared_ptr?

Reply:
Because the csvReader::ReadNext takes in a shared_ptr to recordBatch only